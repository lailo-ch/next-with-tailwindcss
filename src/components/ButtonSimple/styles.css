.btn {
  @apply px-4 py-2;
  @apply text-white;
  @apply bg-green-700;

  &:hover {
    @apply bg-green-800;
  }

  &:disabled {
    @apply bg-gray-300;
    @apply text-gray-500;
    @apply cursor-not-allowed;
  }

  &.loading {
    @apply cursor-wait;
  }

  &.secondary {
    @apply bg-blue-700;

    &:hover {
      @apply bg-blue-800;
    }

    &:disabled {
      @apply bg-gray-300;
      @apply text-gray-500;
      @apply cursor-not-allowed;
    }
  }
}
