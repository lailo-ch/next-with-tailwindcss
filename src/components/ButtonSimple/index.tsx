import * as React from 'react'
import cn from 'classnames'
import './styles.css'

interface Props {
  type?: 'primary' | 'secondary'
  className?: string | object
  loading?: boolean
  [key: string]: any
}

const ButtonSimple: React.FunctionComponent<Props> = ({
  children,
  className,
  type = 'primary',
  loading = false,
  ...props
}) => {
  const disabled = props.disabled || loading

  const mergedClassName = cn(
    'btn',
    {
      secondary: type === 'secondary',
      loading,
    },
    className
  )

  return (
    <button className={mergedClassName} disabled={disabled} {...props}>
      {children}
      {loading && '...'}
    </button>
  )
}

export default ButtonSimple
