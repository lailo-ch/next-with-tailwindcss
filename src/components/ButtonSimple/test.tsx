import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ButtonSimple from '.'

describe('ButtonSimple', () => {
  it('renders with correct label', () => {
    const wrapper = shallow(<ButtonSimple>Hello</ButtonSimple>)
    expect(wrapper.text()).toEqual('Hello')
  })

  it('matches primary (default) type snapshot', () => {
    const wrapper = shallow(<ButtonSimple>Primary</ButtonSimple>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches secondary type snapshot', () => {
    const wrapper = shallow(
      <ButtonSimple type="secondary">Secondary</ButtonSimple>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches with external className as string', () => {
    const wrapper = shallow(
      <ButtonSimple className="external-classname">Hello</ButtonSimple>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches with external className as object', () => {
    const wrapper = shallow(
      <ButtonSimple
        className={{
          'external-classname-true': true,
          'external-classname-false': false,
        }}
      >
        Hello
      </ButtonSimple>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches disabled state', () => {
    const wrapper = shallow(<ButtonSimple disabled>Disabled</ButtonSimple>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches loading state', () => {
    const wrapper = shallow(<ButtonSimple loading>Disabled</ButtonSimple>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
