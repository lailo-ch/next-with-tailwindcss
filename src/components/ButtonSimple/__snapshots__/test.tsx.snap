// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSimple matches disabled state 1`] = `
<button
  className="btn"
  disabled={true}
>
  Disabled
</button>
`;

exports[`ButtonSimple matches loading state 1`] = `
<button
  className="btn loading"
  disabled={true}
>
  Disabled
  ...
</button>
`;

exports[`ButtonSimple matches primary (default) type snapshot 1`] = `
<button
  className="btn"
  disabled={false}
>
  Primary
</button>
`;

exports[`ButtonSimple matches secondary type snapshot 1`] = `
<button
  className="btn secondary"
  disabled={false}
>
  Secondary
</button>
`;

exports[`ButtonSimple matches with external className as object 1`] = `
<button
  className="btn external-classname-true"
  disabled={false}
>
  Hello
</button>
`;

exports[`ButtonSimple matches with external className as string 1`] = `
<button
  className="btn external-classname"
  disabled={false}
>
  Hello
</button>
`;
